pkg/globalnet/controllers/global_ingressip_controller.go

```diff
@@ -70,22 +77,12 @@ func NewGlobalIngressIPController(config *syncer.ResourceSyncerConfig, pool *ipa
 		gip := &submarinerv1.GlobalIngressIP{}
 		_ = runtime.DefaultUnstructuredConverter.FromUnstructured(obj.Object, gip)
 
-		var target string
-		if gip.Spec.Target == submarinerv1.ClusterIPService {
-			target = gip.GetAnnotations()[kubeProxyIPTableChainAnnotation]
-		} else if gip.Spec.Target == submarinerv1.HeadlessServicePod {
-			target = gip.GetAnnotations()[headlessSvcPodIP]
-		}
-
-		if target == "" {
-			continue
-		}
-
 		// nolint:wrapcheck  // No need to wrap these errors.
 		err = controller.reserveAllocatedIPs(federator, obj, func(reservedIPs []string) error {
 			if gip.Spec.Target == submarinerv1.ClusterIPService {
-				return controller.iptIface.AddIngressRulesForService(reservedIPs[0], target)
+				return controller.ensureInternalServiceExists(gip)
```

Comment:
Is this necessary on startup? We only save the allocated IP in the `GlobalIngressIP` if the internal service was successfully created so we **shouldn't** run into the case where the expected internal service is missing on startup. Also if the service was missing then we'd return an error and abort startup with no recovery. If the concern is an unexpected out-of-band delete of the service (eg user "shooting themselves in the foot") then I think we should handle that in the control loop, if at all.

Reply:
> Is this necessary on startup? We only save the allocated IP in the `GlobalIngressIP` if the internal service was successfully created so we **shouldn't** run into the case where the expected internal service is missing on startup. 

In an ideal scenario, the internal service will be present, but if the user accidentally (or intentionally) deletes the submariner-internalService, we will be clearing the AllocatedGlobalIP from the GlobalIngressIP and update its status accordingly.

> Also if the service was missing then we'd return an error and abort startup with no recovery. 

I don't think we will abort the startup. We will clear the AllocatedGlobalIP and update the status of giip - https://github.com/submariner-io/submariner/blob/62927d92c2d4c387bd6c2fe6b62ca7a6acb136d7/pkg/globalnet/controllers/base_controllers.go#L116

> If the concern is an unexpected out-of-band delete of the service (eg user "shooting themselves in the foot") then I think we should handle that in the control loop, if at all.

Yes, if the user deletes the submariner-internalService, we are handling that in the service_controller.go -> onDelete method.

Reply:
> I don't think we will abort the startup. We will clear the AllocatedGlobalIP and update the status of giip

Yes you're right, I missed that. Actually I wrote it a while back - it's hard to remember every detail 😄